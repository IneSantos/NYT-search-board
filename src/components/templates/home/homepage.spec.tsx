import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { BrowserRouter as Router } from "react-router-dom";
import HomePage from "./homepage";

const mockedArticles = [
  {
    id: "",
    headline: "",
    abstract: "",
    keywords: [],
    section: "",
    snippet: "",
    pub_date: "",
    source: "",
    web_url: "",
  },
];
let store;
let mockedStore;

describe("HomePage component tests", () => {
  test(`HomePage component render`, () => {
    mockedStore = configureStore();
    store = mockedStore({
      article: {
        articleList: [
          {
            id: "",
            headline: "",
            abstract: "",
            keywords: [],
            section: "",
            snippet: "",
            pub_date: "",
            source: "",
            web_url: "",
          },
        ],
        page: 0,
        searchClicked: false,
      },
    });

    const { container } = render(
      <Router>
        <Provider store={store}>
          <HomePage
            articles={mockedArticles}
            searchInput="This is an input"
            handleSearch={jest.fn()}
          />
        </Provider>
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
