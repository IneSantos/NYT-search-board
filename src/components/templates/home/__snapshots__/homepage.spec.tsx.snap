// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component tests HomePage component render 1`] = `
<section>
  <header>
    <div
      class="banner"
    >
      “The New York Times” article search application
    </div>
  </header>
  <div
    class="page-body-wrapper"
  >
    <div
      class="page-body"
    >
      <div
        class="header-wrapper"
      >
        <div
          class="search-wrapper"
        >
          <h2>
            Type search query term in here: 
          </h2>
          <div
            class="search"
          >
            <input
              class="search-term"
              placeholder="What are you looking for?"
              type="text"
              value="This is an input"
            />
            <button
              class="clear-button"
              type="reset"
            >
              clear
            </button>
            <button
              class="search-button"
              type="submit"
            >
              <i
                class="fas fa-search"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="articles-list"
      >
        <h2>
          Results:
        </h2>
        <a
          href="/article/0"
          style="text-decoration: none;"
        >
          <div
            class="card-wrapper"
          >
            <div
              class="card-header"
            >
              <h3
                class="title-small"
              />
            </div>
          </div>
        </a>
      </div>
      <div
        class="pagination"
      >
        <a
          class="article-action article-action-next-page"
        >
           
          Next page &gt;
           
        </a>
        <a
          class="article-action article-action-prev-page"
        >
           
          &lt; Prev page
           
        </a>
      </div>
    </div>
  </div>
</section>
`;
